src/ext/ddtrace_extra.c

```diff
@@ -0,0 +1,118 @@
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <SAPI.h>
+#include <Zend/zend.h>
+#include <Zend/zend_exceptions.h>
+#include <php.h>
+#include <php_ini.h>
+#include <php_main.h>
+#include <ext/spl/spl_exceptions.h>
+#include <ext/standard/info.h>
+
+#include "compatibility.h"
+#include "ddtrace.h"
+#include "ddtrace_extra.h"
+#include "debug.h"
+#include "env_config.h"
+#include "serializer.h"
+
+#if PHP_VERSION_ID < 70000
+typedef int32_t zend_long;
+#endif
+
+ZEND_EXTERN_MODULE_GLOBALS(ddtrace);
+
+/* {{{ proto string dd_trace_serialize_msgpack(array trace_array) */
+PHP_FUNCTION(dd_trace_serialize_msgpack) {
+    PHP5_UNUSED(return_value_used, this_ptr, return_value_ptr, ht);
+    PHP7_UNUSED(execute_data);
+
+    if (DDTRACE_G(disable)) {
+        RETURN_BOOL(0);
+    }
+
+    zval *trace_array;
+
+    if (zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "a", &trace_array) == FAILURE) {
+        if (DDTRACE_G(strict_mode)) {
+            zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0 TSRMLS_CC, "Expected an array");
+        }
+        RETURN_BOOL(0);
+    }
+
+    if (ddtrace_serialize_simple_array(trace_array, return_value TSRMLS_CC) != 1) {
+        RETURN_BOOL(0);
+    }
+} /* }}} */
+
+// method used to be able to easily breakpoint the execution at specific PHP line in GDB
+PHP_FUNCTION(dd_trace_noop) {
+    PHP5_UNUSED(return_value_used, this_ptr, return_value_ptr, ht);
+    PHP7_UNUSED(execute_data);
+
+    if (DDTRACE_G(disable)) {
+        RETURN_BOOL(0);
+    }
+
+    RETURN_BOOL(1);
+}
+
+static zend_long get_memory_limit(TSRMLS_D) {
+    char *raw_memory_limit = ddtrace_get_c_string_config("DD_MEMORY_LIMIT");
```

Comment:
Is there any reason why we're using the `DD_` prefix here instead of the `DD_TRACE_` prefix? I don't know if we narrowed down a distinction so just wanted to double check. :)

Reply:
Oh I haven't thought too much about it. `DD_TRACE_` does feel more canonical

Reply:
Yeah, me neither so it got me thinking about why we're choosing one over the other... I think I hear an RFC idea... *jots down TODO*. :)